src/sagemaker/image_uris.py

```diff
@@ -106,6 +112,8 @@ def retrieve(
             distribution,
             base_framework_version,
             training_compiler_config,
+            tolerate_vulnerable_model,
```

Comment:
Apologies for not thinking about this earlier, but here, should we pass `tolerate_vulnerable_model=True` and remove the argument from the utility function signature ?

The idea is that only the script may have a vulnerable flag, not the image itself. So if a customer wants to invoke this utility, but use their own scripts URI, should we block them? The same goes for the model artifact URI. Let's discuss, this is an important point.

Reply:
Good point. After discussing on Slack, we determined that the model and version should be entirely vulnerable for all of inference or training. This means that accessing model artifact URIs or image URIs which corresponds to a vulnerable script URI should also raise an Exception.